File: app/src/main/java/com/jyotirmayg/mytestproject/feature/main/MainViewModel.java
package com.jyotirmayg.mytestproject.feature.main;

import android.content.Context;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.jyotirmayg.mytestproject.data.db.entities.Item;
import com.jyotirmayg.mytestproject.data.repositories.ItemRepository;

import java.util.List;

/**
 * @author jyoti
 * @created on 17-05-2022
 */
public class MainViewModel extends ViewModel {

    private final ItemRepository itemRepository;

    public MainViewModel(Context context) {
        itemRepository = ItemRepository.getInstance(context);
    }

    /**
     * This is called to get updated list from the database.
     */
    public LiveData<List<Item>> getItemList() {
        return itemRepository.getUpdatedItemList();
    }

}

File: app/src/main/java/com/jyotirmayg/mytestproject/view/main/MainViewModel.kt
package com.jyotirmayg.mytestproject.view.main

import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import androidx.navigation.NavDirections
import com.jyotirmayg.mytestproject.base.Application
import com.jyotirmayg.mytestproject.data.db.entities.Item
import com.jyotirmayg.mytestproject.data.repositories.ItemRepository
import kotlinx.coroutines.launch

/**
 * @author jyoti
 * @created on 14-05-2022
 */
class MainViewModel(app: Application) : AndroidViewModel(app) {

    // for navigation
    private val _navigationFlow = MutableLiveData<Route>()
    val mainNavigationFlow: LiveData<Route> = _navigationFlow

    /**
     * This will be called to route from one fragment to another fragment.
     * @param route where to route.
     * */
    fun routeTo(route: Route) {
        _navigationFlow.postValue(route)
    }
}